plugins/DotToc.ts

```diff
@@ -0,0 +1,46 @@
+import visit from 'unist-util-visit';
+import { Node } from 'unist';
+
+export let toc: NodeTableContent[] = [];
```

Comment:
This approach of building it this way is weird... is this the only way?

Reply:
I was looking for a better way to do what we want, but there doesn't seem to be an official way to extract a part of the content, the table of contents, in this case, so after reading on Reddit I found out this way to do it, but I keep looking for a better way.